src/components/CreateTemplate/index.tsx

```diff
@@ -0,0 +1,228 @@
+/* eslint-disable require-jsdoc */
+import { useStoreSelector } from "@/store/hooks";
+import {
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  FormControl,
+  FormLabel,
+  Input,
+  Box,
+  Flex,
+  Text,
+  Textarea,
+  Icon,
+  Badge,
+  useDisclosure,
+} from "@chakra-ui/react";
+import React, { useRef, useState } from "react";
+import { FaUserCircle } from "react-icons/fa";
+import { useAddTemplateMutation } from "@/redux/api/templateApi";
+import SuccessNotice from "./SuccessNotice";
+import generateNewTemplate from "@/utils/generateNewTemplate";
+import { userNameEllip } from "../../utils/handleLongUserName";
+
+interface Props {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const maxCourtNameLen = 15;
+const maxDescriptionLen = 200;
+const maxUserNameDisplay = 20;
+const regex = /[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~]/g;
+
+const inputErrorInit = {
+  courtNameFullErr: false,
+  courtNameNullErr: false,
+  descriptionOverLimit: false,
+};
+
+const inputErrorMsg = {
+  nameFullErrMsg: `Court name cannot have more than ${maxCourtNameLen} characters`,
+  nameNullErrMsg: "Court name cannot be empty",
+  descriptionLenErrMsg: `Description can not over ${maxDescriptionLen} words`,
+};
+
+function CreateTemplate({ isOpen, onClose }: Props) {
+  const courtNameRef = useRef<HTMLInputElement>(null);
+  const descriptionRef = useRef<HTMLTextAreaElement>(null);
+  const [inputError, setInputError] = useState(inputErrorInit);
+  const [textAreaLen, setTextAreaLen] = useState(0);
+  const { userId, firstName, lastName } = useStoreSelector((state) => state.user);
+  const { activeCourt: selectedCourt } = useStoreSelector((state) => state.courtSpecData);
+  const { court: selectedCourtTileData } = useStoreSelector((state) => state.tile.present);
+  const [addTemplate] = useAddTemplateMutation();
+  const courtType = "basketball";
+  const userFullName = userNameEllip(`${firstName} ${lastName}`, maxUserNameDisplay);
```

Comment:
Is the requirement here limited to display only 20 characters, or the part that exceeds the width will not be displayed, because this can be implemented with css